qubesmanager/settings.py

```diff
@@ -627,7 +627,15 @@ def check_mem_changes(self):
             self.warn_too_much_mem_label.setVisible(True)
 
     def check_warn_templatenetvm(self):
-        if self.vm.klass == 'TemplateVM':
+        if self.vm.klass != 'TemplateVM':
+            return
+
+        current_netvm = self.netVM.currentData()
+        if current_netvm is None:
+            return
+
+        default_netvm = self.vm.property_get_default('netvm')
+        if current_netvm != qubesadmin.DEFAULT or default_netvm:
```

Comment:
Did you meant `current_netvm != default_netvm`?

Reply:
I am trying to check:

Either `current_netvm` is _not_ set to `(default)` on the Qube Manager, or (if it is then) `default_netvm` is not `None`

If this check succeeds, then the VM has networking enabled.

If you have any suggestions on making the check cleaner, I'd be happy to change it.

Reply:
Simply checking `current_netvm != default_netvm` wouldn't work in all cases.

Consider the case when `default_netvm` is not set to `None` in global settings, say it's set to `vm-1`. Then the check wouldn't trigger when we set a template's NetVM to `vm-1` although the template would indeed have networking enabled.

Reply:
The default is always for TemplateVMs to have no network, regardless of what `default_netvm` is set to.

Reply:
Yes, the default NetVM for TemplateVMs is set to `(none)` in the Qube Manager. But that's different from setting it to `(default)`, which is the `default_netvm`.

In this check, I essentially avoid a spurious warning if a user changes the NetVM to `(default)` and `default_netvm` is set to `(none)` in global settings.

Do you think this is reasonable?

Reply:
For TemplateVMs, `(none)` and `(default)` *should* be equivalent, regardless of what `default_netvm` is.

Reply:
I'm afraid that they don't seem to be equivalent.

Here is what I see on my system:
`(default)` takes the value set in the global settings, and shows up as `default (vm-1)` for example, if `vm-1` is the `default_netvm` from global settings.

Reply:
I actually don't think they should be equivalent, because that would mean `(default)` has a different meaning for TemplateVMs (always `(none)`), than AppVM (`default_netvm`).

Reply:
> I actually don't think they should be equivalent, because that would mean `(default)` has a different meaning for TemplateVMs (always `(none)`), than AppVM (`default_netvm`).

It actually does have a different meaning, precisely because setting a NetVM for a TemplateVM is almost always a bad idea.

Reply:
Right, I see your point.

I guess the question here is, currently `(none)` and `(default)` are not equivalent. Should we make them equivalent?

I am happy to remove the check from my PR if we decide to do so.

Reply:
Internally, `(none)` maps to `netvm` being set to the empty string (which means “no NetVM”), while `(default)` means that `netvm` is not set at all.  An unset property is interpreted as having a default value.  For TemplateVMs, the default is networking disabled, while for other types of VMs, the default is set by the `default_netvm` global setting.

In short, both `(none)` and `(default)` are okay.  Anything else should give a warning.

Reply:
Thanks.

So I should simplify the check to just `if current_netvm != qubesadmin.DEFAULT:`, right?

Reply:
> Thanks.
> 
> So I should simplify the check to just `if current_netvm != qubesadmin.DEFAULT:`, right?

Exactly

Reply:
Fixed as suggested.